### ISSUE TITLE
Unexpected token .. when using rest parameter in destructured assignment

### ISSUE THREAD

**Issue by pbeshai** [eslint/eslint#2352]
With eslint v0.19.0, I get an error with the following code:

eslintrc
```json
{ 
    "ecmaFeatures": {
        "destructuring": true,
        "spread": true,
        "restParams": true 
    }
}
```

test.js
```js
const { header, ...other } = { header: 'foo', bar: 1, baz: 2 };
```

`eslint -c eslintrc test.js` produces the output:
```
test.js
  1:17  error  Unexpected token ..

✖ 1 problem (1 error, 0 warnings)
```

**Comment by lo1tuma** [comment]
AFAIK this is not supported in ES6. It works only for arrays but not for objects.

**Comment by nzakas** [comment]
You are correct. What you are doing here is spread properties operator, which is a proposal for ES7, but not yet standardized.

**Comment by pbeshai** [comment]
Oh interesting because React recommends it as the standard way to transfer props to components (https://facebook.github.io/react/docs/transferring-props.html).   ... And upon closer inspection they do call it out as experimental ES7. Thanks!

**Comment by rpominov** [comment]
Since this already [supported by Babel](https://babeljs.io/repl/#?experimental=false&evaluate=false&loose=false&spec=false&playground=false&code=a%20%3D%20%7B...b%2C%20text%7D) perhaps it's time to add support to ESLint?

Please, reopen.

**Comment by hzoo** [comment]
@rpominov you can use https://github.com/babel/babel-eslint

**Comment by rpominov** [comment]
Yep, was already using it but wrote `(old) => {...old, x}` when should have `(old) => ({...old, x})`.

Thanks for the response, anyway!

**Comment by ilyavolodin** [comment]
@rpominov ESLint doesn't support experimental features. That's what babel-eslint is for. Until es7 is approved you will have to use babel for this.

**Comment by rpominov** [comment]
@ilyavolodin Right, I just didn't get it at first. Thanks for the clarification anyway :+1:
